### ISSUE TITLE
A11y keyboard focus control in the Button Toggle Component

### ISSUE THREAD

**Issue by DaveBest99** [angular/components#3704]
#### Bug, feature request, or proposal:
Bug:
The demo does not include all elements in the Tab focus order, and automatically announce screen changes.

#### What is the expected behavior?
Whenever a user keyboard action causes a screen update to occur, that information should be automatically announced by the screen reader.

#### What is the current behavior?
Currently the "Selected value:" is updated as the Radio Buttons are selected, but is not automatically announced by the screen reader.

#### What are the steps to reproduce?
See below.

Providing a Plunker (or similar) is the *best* way to get the team to see your issue.
Plunker template: https://goo.gl/DlHd6U

#### What is the use-case or motivation for changing an existing behavior?
Create a good screen reader user experience.

#### Which versions of Angular, Material, OS, browsers are affected?
Using screen reader JAWS17 with browsers Firefox, Internet Explorer 11, and Google Chrome, on Windows10.

#### Is there anything else we should know?
Screen reader user experience follows:

1. Press Tab key to the Button demo component in the Tab Panel.
JAWS says "format_align_left button checked 1 of 4", and the braille display shows "format_align_left rbtn 1 of 4".
Note, Arrow keys allow the user to browse the page Button and Image elements, as expected.

2. Press Enter key on the Radio Button to access the Radio Button group.
The "format_align_justify" Radio Button is Disabled and does not get focus when Arrow keys rotate through the Radio Button group, as expected.
However, the message "Selected value: left" is not in the Tab focus order, and is not automatically announced when the status of the Radio Button changes with each press of the Arrow keys. Note, this may be a feature of the demo, and not a defect with the Button Module, and if so, then there is no problem with the Button Module.

3. Note, this page demonstrates a Radio Button Group, rather than a Toggle Button. A Toggle button is a two-state button that can be either off (not pressed) or on (pressed). To tell assistive technologies that a button is a toggle button, specify a value for the attribute aria-pressed. For example, a button labeled mute in an audio player could indicate that sound is muted by setting the pressed state true. Important: it is critical the label on a toggle does not change when its state changes. In this example, when the pressed state is true, the label remains "Mute" so a screen reader would say something like "Mute toggle button pressed". Alternatively, if the design were to call for the button label to change from "Mute" to "Unmute," the aria-pressed attribute would not be needed. 

However, in the Demo Overview Tab Panel, under the Heading "Basic button-toggles", the 
"Toggle me!" Checkbox behaves as expected when pressing Spacebar or Enter key. There is no accessibility issue with this Toggle Button example.

See:
The Roles Model | Accessible Rich Internet Applications (WAI-ARIA) 1.0
http://w3c.github.io/aria-practices/#grid

WAI-ARIA Authoring Practices 1.1 - 2.6 Button
http://w3c.github.io/aria-practices/#h-dialog_modal#h-button
